File: Lab09/01Lab09_2025_1_20203361_Valentin_James/Proyecto/src/Vehiculo.cpp
//
// Created by james on 27/11/2025.
//

#include "Vehiculo.h"

Vehiculo::Vehiculo(const Vehiculo &orig) {
    *this = orig;
}

Vehiculo &Vehiculo::operator=(const Vehiculo &orig) {
    if (this==&orig)return *this;
    setPlaca(orig.getPlaca());
    setModelo(orig.getMarca());
    setModelo(orig.getModelo());
    return *this;
}

string Vehiculo::getPlaca() const {
    return placa;
}

void Vehiculo::setPlaca(const string &placa) {
    this->placa = placa;
}

string Vehiculo::getMarca() const {
    return marca;
}

void Vehiculo::setMarca(const string &marca) {
    this->marca = marca;
}

string Vehiculo::getModelo() const {
    return modelo;
}

void Vehiculo::setModelo(const string &modelo) {
    this->modelo = modelo;
}

void Vehiculo::leer(ifstream &arch) {
    getline(arch, placa,',');
    if (arch.eof())return;
    getline(arch, marca,',');
    getline(arch, modelo);
}

void Vehiculo::imprime(ofstream &arch) const{

}

ifstream &operator>>(ifstream &input, Vehiculo &v) {
    v.leer(input);
    return input;
}

ofstream &operator<<(ofstream &output, const Vehiculo &v) {
    v.imprime(output);
    return output;
}

File: Lab10/01Lab10_2025_1_20203361_Valentin_James/Proyecto/Lab10_25_1/src/Vehiculo.java

import java.util.Scanner;

/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

/**
 *
 * @author james
 */
public class Vehiculo extends Registro{
    private String placa;
    private String marca;
    private String modelo;
    private int anhoFb;
    private String categoria;
    private Propietario propietario;
    
    @Override
    public void cargar(Scanner archivo){
        String texto =archivo.next();
        if(texto.equals("FIN"))return;
        this.placa=texto;
        this.marca=archivo.next();
        this.modelo=archivo.next();
        this.anhoFb=archivo.nextInt();
        this.categoria=archivo.next();
    }
    
    @Override
    public void imprimir(){
        System.out.println(this);
    }
    
    @Override
    public String toString(){
        return "Placa: "+ placa
                + "\n Marca: "+ marca 
                + "\n Modelo: " + modelo 
                + "\n Año de Fabricación: "+ anhoFb 
                + "\n Categoría: " + categoria 
                + "\n";
    }
    
    public String getPlaca() {
        return placa;
    }

    public void setPlaca(String placa) {
        this.placa = placa;
    }

    public String getMarca() {
        return marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public int getAnhoFb() {
        return anhoFb;
    }

    public void setAnhoFb(int anhoFb) {
        this.anhoFb = anhoFb;
    }

    public String getCategoria() {
        return categoria;
    }

    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }

    public Propietario getPropietario() {
        return propietario;
    }

    public void setPropietario(Propietario propietario) {
        this.propietario = propietario;
    }
}
